### PR TITLE
Add timeout for running pytest (resolves #1735)

### LINKED ISSUE #1735: Timeout tests with PyTest instead of Jenkins for better debugging hanging tests
Currently, if a test hangs Jenkins eventually shuts it down. Unfortunately this leaves no indication of which test was running or what may have been causing it to hang.

By making tests timeout from [PyTest timeout](https://pypi.python.org/pypi/pytest-timeout) we can get a full trace as well as information on which test was hanging.

This will also probably be enough to fix #1700.

Idea credit to [Joel](https://github.com/BD2KGenomics/toil/pull/1731#issuecomment-310813970).

### PR DESCRIPTION
**Pull request by jessebrennan**
This amends the call to `pytest` in run_test.py by adding the option `--timeout=600` which gives any particular test 10 minutes to run. I imagine this may be enough time for all of the tests. If it is not, we can increase the overall timeout by adjusting the option above, or any particularly lengthy tests can be given their own timeout by adding the decorator `@pytest.mark.timeout(XXX)` where `XXX` is some number of seconds.

This passes tests locally, but pytest-timeout may need to be installed on Jenkins for this to work.

### CODE REVIEW

**Comment by jessebrennan** [comment]
Jenkins, test this please.

**Comment by jessebrennan** [comment]
Boto throws 400 error. Let's try again. Jenkins, test this please.

**Comment by jessebrennan** [comment]
@ejacox I moved this to sprint 7. It is essentially done—just waiting for tests to pass since boto has been throwing 400s. Getting this in the next release will make future development much easier.

File: run_tests.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
 
 def run_tests(keywords, index, args):
-    args = [sys.executable, '-m', 'pytest', '-vv',
+    args = [sys.executable, '-m', 'pytest', '-vv', '--timeout=600',
```
Comment (ejacox) [inline run_tests.py]: What does setting this value do?

Comment (ejacox) [inline run_tests.py]: Is it automatically read by pytest?

**Comment by jessebrennan** [comment]
@ejacox this is ready to merge!

File: run_tests.py (hunk 1 above)
Comment (jessebrennan) [inline run_tests.py]: Yep! see [here]( https://pypi.python.org/pypi/pytest-timeout).

**Pull request merged**